apps/api/src/app/workflows-v2/usecases/generate-preview/generate-preview.usecase.ts

```diff
@@ -1,68 +1,156 @@
+import { pathsToObject } from '../../util/path-to-object';
 
 @Injectable()
 export class GeneratePreviewUsecase {
   constructor(
     private legacyPreviewStepUseCase: PreviewStep,
-    private prepareAndValidateContentUsecase: PrepareAndValidateContentUsecase,
-    private buildStepDataUsecase: BuildStepDataUsecase
+    private buildStepDataUsecase: BuildStepDataUsecase,
+    private getWorkflowByIdsUseCase: GetWorkflowByIdsUseCase
   ) {}
 
   async execute(command: GeneratePreviewCommand): Promise<GeneratePreviewResponseDto> {
     const dto = command.generatePreviewRequestDto;
     const stepData = await this.getStepData(command);
+    const workflow = await this.findWorkflow(command);
 
-    const validatedContent: ValidatedContentResponse = await this.getValidatedContent(dto, stepData, command.user);
+    const variablesExample = this.buildVariablesExample(workflow, stepData, dto);
     const executeOutput = await this.executePreviewUsecase(
       command,
       stepData,
-      validatedContent.finalPayload,
-      validatedContent.finalControlValues
+      variablesExample,
+      dto.controlValues || {}
     );
 
     return {
-      issues: validatedContent.issues,
       result: {
         preview: executeOutput.outputs as any,
         type: stepData.type as unknown as ChannelTypeEnum,
       },
-      previewPayloadExample: validatedContent.finalPayload,
-    };
+      previewPayloadExample: variablesExample,
+    } as GeneratePreviewResponseDto;
   }
 
-  private async getValidatedContent(dto: GeneratePreviewRequestDto, stepData: StepDataDto, user: UserSessionData) {
-    if (!stepData.controls?.dataSchema) {
-      throw new StepMissingControlsException(stepData.stepId, stepData);
+  private buildVariablesExample(
+    workflow: WorkflowInternalResponseDto,
+    stepData: StepDataDto,
+    dto: GeneratePreviewRequestDto
+  ) {
+    const variablesExample = this.generateVariablesExample(dto, stepData);
+
+    if (workflow.origin === WorkflowOriginEnum.EXTERNAL) {
+      variablesExample.payload = this.generateSamplePayload(workflow.payloadSchema);
     }
 
-    return await this.prepareAndValidateContentUsecase.execute({
-      stepType: stepData.type,
-      controlValues: dto.controlValues || stepData.controls.values,
-      controlDataSchema: stepData.controls.dataSchema,
-      variableSchema: stepData.variables,
-      previewPayloadFromDto: dto.previewPayload,
-      user,
+    return _.merge(variablesExample, dto.previewPayload as Record<string, unknown>);
+  }
+
+  private generateVariablesExample(dto: GeneratePreviewRequestDto, stepData: StepDataDto) {
+    const controlValues = Object.values(dto.controlValues || stepData.controls.values).join('');
+    const variablesExample = pathsToObject(extractTemplateVars(controlValues), {
+      valuePrefix: '{{',
+      valueSuffix: '}}',
     });
+
+    return variablesExample;
+  }
+
+  private buildPayloadExample(
```

Comment:
Is this the duplication we talked about?

Reply:
not sure, which duplication you refer to but this code is old leftovers and not used anymore i removed it.